google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/gcs/GcsBucketContentEditorPanelTest.java

```diff
@@ -291,6 +292,38 @@ public void testBlobSizeDisplay_GB() {
     assertThat(bucketTable.getValueAt(0, COL_NAME_TO_INDEX.get("Size"))).isEqualTo("100.0 GB");
   }
 
+  @Test
+  public void testBucketListException_showsErrorMessage() {
+    when(bucketVirtualFile.getBucket().list(any(BlobListOption.class), any(BlobListOption.class)))
+        .thenThrow(StorageException.class);
+
+    editorPanel = new GcsBucketContentEditorPanel(bucketVirtualFile.getBucket());
+    editorPanel.initTableModel();
+
+    JTable bucketTable = editorPanel.getBucketContentTable();
+    assertThat(bucketTable.getColumnCount()).isEqualTo(0);
+    assertThat(bucketTable.getRowCount()).isEqualTo(0);
+    assertFalse(editorPanel.getNoBlobsPanel().isVisible());
+    assertFalse(editorPanel.getLoadingPanel().isVisible());
+    assertTrue(editorPanel.getErrorPanel().isVisible());
+  }
+
+  @Test
+  public void testErrorMessageIsCleared_afterSuccessfulBucketList() {
+    when(bucketVirtualFile.getBucket().list(any(BlobListOption.class), any(BlobListOption.class)))
+        .thenThrow(StorageException.class);
+
+    editorPanel = new GcsBucketContentEditorPanel(bucketVirtualFile.getBucket());
+    editorPanel.initTableModel();
+
+    assertTrue(editorPanel.getErrorPanel().isVisible());
+
+    // Re-initialize mocks so the exception is not thrown
+    GcsTestUtils.setupVirtualFileWithBucketMocks(bucketVirtualFile);
+
+    assertFalse(editorPanel.getErrorPanel().isVisible());
```

Comment:
Don't you have to make a call to `updateTableModel()` or something to retry the list call?

Reply:
good eye. I'm in the process of fixing this test now

Reply:
fixed